code_generation/quantity.py

```diff
@@ -26,23 +29,32 @@ def reserve_scope(self, scope):
             raise Exception
 
     def get_leaf(self, shift, scope):
-        if shift in self.get_shifts(scope):
-            return self.name + shift
-        return self.name
+        log.debug("{} - getting shift {} for scope {}".format(self.name, shift, scope))
+        leaf = self.name
+        if "global" in self.shifts.keys():
+            if shift in self.shifts["global"].keys():
+                leaf = self.shifts["global"][shift]
+        elif scope in self.shifts.keys():
+            if shift in self.shifts[scope].keys():
+                leaf = self.shifts[scope][shift]
+        return leaf
 
     def get_leafs_of_scope(self, scope):
-        return [self.name] + [self.name + shift for shift in self.get_shifts(scope)]
+        result = [self.name] + [
+            self.shifts[scope][shift] for shift in self.get_shifts(scope)
```

Comment:
I think this does not work correctly. get_shifts should give you all shifts in global and scope but you try to look them up in self.shifts[scope].
Instead of directly accessing the self.shifts dict you could use get_leaf(...) in the loop.